In [31]:
import pandas as pd
import numpy as np

from scipy.stats import ttest_1samp
from scipy.stats import ttest_ind
from scipy.stats import binom

In [16]:
df = pd.read_csv('property.csv')
df.head()

Suburb           Address  Rooms Type      Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB  Nelson   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
1  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
2  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
3  4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0           NaN   
4  4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN        Yarra  -37.7996    144.9984  Northern Metropolitan   
1     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
2     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   
3        NaN        Yarra  -37.7969    144.9969  Northern Metropolitan   
4     2014.0        Yarra  -37.8072    144.9941  Northern Metropolitan   

  Propertycount  
0        4019.0  
1        4019.0  
2        4019.0  
3        4019.0  
4        4019.0  

[5 rows x 21 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  float64
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

In [6]:
df.describe()

Rooms         Price      Distance      Postcode      Bedroom2  \
count  13580.000000  1.358000e+04  13580.000000  13580.000000  13580.000000   
mean       2.937997  1.075684e+06     10.137776   3105.301915      2.914728   
std        0.955748  6.393107e+05      5.868725     90.676964      0.965921   
min        1.000000  8.500000e+04      0.000000   3000.000000      0.000000   
25%        2.000000  6.500000e+05      6.100000   3044.000000      2.000000   
50%        3.000000  9.030000e+05      9.200000   3084.000000      3.000000   
75%        3.000000  1.330000e+06     13.000000   3148.000000      3.000000   
max       10.000000  9.000000e+06     48.100000   3977.000000     20.000000   

           Bathroom           Car       Landsize  BuildingArea    YearBuilt  \
count  13580.000000  13518.000000   13580.000000   7130.000000  8205.000000   
mean       1.534242      1.610075     558.416127    151.967650  1964.684217   
std        0.691712      0.962634    3990.669241    541.014538    37.273762   
min        0.000000      0.000000       0.000000      0.000000  1196.000000   
25%        1.000000      1.000000     177.000000     93.000000  1940.000000   
50%        1.000000      2.000000     440.000000    126.000000  1970.000000   
75%        2.000000      2.000000     651.000000    174.000000  1999.000000   
max        8.000000     10.000000  433014.000000  44515.000000  2018.000000   

          Lattitude    Longtitude  Propertycount  
count  13580.000000  13580.000000   13580.000000  
mean     -37.809203    144.995216    7454.417378  
std        0.079260      0.103916    4378.581772  
min      -38.182550    144.431810     249.000000  
25%      -37.856822    144.929600    4380.000000  
50%      -37.802355    145.000100    6555.000000  
75%      -37.756400    145.058305   10331.000000  
max      -37.408530    145.526350   21650.000000

1. For the suburb Altona, it is postulated that a typical property sells for $800,000. Use the data at hand to test this assumption. Is the typical property price really $800,000 or has it increased? Use a significance level of 5%.

In [13]:
t_stat, p_value = ttest_1samp(df.loc[df['Suburb'] == 'Altona']['Price'], 800000)
print(t_stat, p_value)

1.0277020770199676 0.307483271305555


Since P value is less tha sigificance value, we dont have enough evidence to conclude that the typical property price really has increased $800,000 

2. For the year 2016, is there any difference in prices of properties sold in the summer months vs winter months? Consider months from October till March as winter months and rest as summer months. Use a significance level of 5%.

In [23]:
df['Date'] = pd.to_datetime(df['Date'], format="%d/%m/%Y")
df['Year'] = df['Date'].dt.year
df['Month'] = df['Date'].dt.month

In [26]:
df_2016 = df[df['Year'] == 2016]
df_2016

Suburb           Address  Rooms Type      Price Method  \
0       Abbotsford      85 Turner St      2    h  1480000.0      S   
1       Abbotsford   25 Bloomburg St      2    h  1035000.0      S   
4       Abbotsford       55a Park St      4    h  1600000.0     VB   
5       Abbotsford    129 Charles St      2    h   941000.0      S   
6       Abbotsford      124 Yarra St      3    h  1876000.0      S   
...            ...               ...    ...  ...        ...    ...   
7426    Gardenvale  7/32 Gardenia Rd      2    u   440000.0      S   
7427    Gardenvale  1/18 Magnolia Rd      2    u   595000.0      S   
7428    Gardenvale    25 Magnolia Rd      4    h  2200000.0      S   
7429  Princes Hill      76 Arnold St      3    h  1700000.0     VB   
7430  Princes Hill  77 McIlwraith St      4    h  1908000.0      S   

            SellerG       Date  Distance  Postcode  ...  Landsize  \
0            Biggin 2016-12-03       2.5    3067.0  ...     202.0   
1            Biggin 2016-02-04       2.5    3067.0  ...     156.0   
4            Nelson 2016-06-04       2.5    3067.0  ...     120.0   
5            Jellis 2016-05-07       2.5    3067.0  ...     181.0   
6            Nelson 2016-05-07       2.5    3067.0  ...     245.0   
...             ...        ...       ...       ...  ...       ...   
7426         Biggin 2016-06-18       9.8    3185.0  ...       0.0   
7427         Biggin 2016-11-19       9.8    3185.0  ...       0.0   
7428  hockingstuart 2016-08-28       9.8    3185.0  ...     474.0   
7429        Collins 2016-11-27       3.7    3054.0  ...     341.0   
7430         Nelson 2016-05-28       3.7    3054.0  ...     256.0   

      BuildingArea  YearBuilt  CouncilArea  Lattitude  Longtitude  \
0              NaN        NaN        Yarra   -37.7996    144.9984   
1             79.0     1900.0        Yarra   -37.8079    144.9934   
4            142.0     2014.0        Yarra   -37.8072    144.9941   
5              NaN        NaN        Yarra   -37.8041    144.9953   
6            210.0     1910.0        Yarra   -37.8024    144.9993   
...            ...        ...          ...        ...         ...   
7426          75.0        NaN    Glen Eira   -37.8971    145.0093   
7427          75.0     1960.0    Glen Eira   -37.8994    145.0101   
7428         230.0     2010.0    Glen Eira   -37.8988    145.0098   
7429           NaN        NaN        Yarra   -37.7832    144.9655   
7430         156.0     1890.0        Yarra   -37.7833    144.9681   

                 Regionname  Propertycount  Year Month  
0     Northern Metropolitan         4019.0  2016    12  
1     Northern Metropolitan         4019.0  2016     2  
4     Northern Metropolitan         4019.0  2016     6  
5     Northern Metropolitan         4019.0  2016     5  
6     Northern Metropolitan         4019.0  2016     5  
...                     ...            ...   ...   ...  
7426  Southern Metropolitan          534.0  2016     6  
7427  Southern Metropolitan          534.0  2016    11  
7428  Southern Metropolitan          534.0  2016     8  
7429  Northern Metropolitan         1008.0  2016    11  
7430  Northern Metropolitan         1008.0  2016     5  

[6336 rows x 23 columns]

In [27]:
summer_months = [4, 5, 6, 7, 8, 9]
winter_months = [10, 11, 12, 1, 2, 3]

In [28]:
summer_data = df_2016[df_2016['Month'].isin(summer_months)]
winter_data = df_2016[df_2016['Month'].isin(winter_months)]

In [30]:
t_stat, p_value = ttest_ind(summer_data['Price'], winter_data['Price'], equal_var=False)
print(t_stat, p_value)

-3.9211012778003753 8.950129747463378e-05


he p value is significantlly less than significance level hence we can conclude that there is a statistically significant difference in property prices between the summer and winter months in 2016.

3. For the suburb of Abbotsford, what is the probability that out of 10 properties sold, 3 will not have car parking? Use the column car in the dataset. Round off your answer to 3 decimal places

In [32]:
abbotsford_data = df[df['Suburb'] == 'Abbotsford']
no_car_proportion = (abbotsford_data['Car'] == 0).mean()
probability = binom.pmf(3, 10, no_car_proportion)
no_car_proportion, round(probability, 3)

(0.26785714285714285, 0.26)

The probability that out of 10 properties sold, 3 will not have car parking is 0.26

4. In the suburb Abbotsford, what are the chances of finding a property with 3 rooms? Round your answer to 3 decimal places.

In [33]:
three_rooms_proportion = (abbotsford_data['Rooms'] == 3).mean()
round(three_rooms_proportion, 3)

0.357

The chances of finding a property with 3 rooms is 0.357

5. In the suburb Abbotsford, what are the chances of finding a property with 2 bathrooms? Round your answer to 3 decimal places.

In [34]:
two_bathrooms_proportion = (abbotsford_data['Bathroom'] == 2).mean()
round(two_bathrooms_proportion, 3)

0.339

The chances of finding a property with 2 bathrooms is 0.339